owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -431,10 +435,9 @@ private void populateDirectoryList() {
         mFile = getStorageManager().getFileByPath(full_path);
         if (mFile != null) {
             Vector<OCFile> files = getStorageManager().getFolderContent(mFile);
-            files = sortFileList(files);
+            mFiles = sortFileList(files);
```

Comment:
I think that you need to update directly the adapter's vector. And after updating, call `notifyDataSetChanged()`. `mFiles` is not needed in this file.